gradle-metric-schema/src/main/java/com/palantir/metric/schema/gradle/MetricSchemaPlugin.java

```diff
@@ -117,9 +120,10 @@ private static SourceDirectorySet createSourceSet(Project project) {
         return sourceSet;
     }
 
-    private static void configureJavaSource(Project project, Provider<Directory> outputDir) {
-        JavaPluginConvention javaPlugin = project.getConvention().findPlugin(JavaPluginConvention.class);
-        javaPlugin.getSourceSets().getByName("main").getJava().srcDir(outputDir);
+    private static SourceSet getMainSourceSet(Project project) {
+        JavaPluginExtension javaPlugin = project.getExtensions().getByType(JavaPluginExtension.class);
+        javaPlugin.withSourcesJar();
```

Comment:
This feels wrong, this plugin shouldn't enable the sources JAR if it's not enabled by another plugin.